src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -1770,617 +1796,1423 @@ HRESULT CorProfiler::RunAutoInstrumentationLoader(const ComPtr<IMetaDataEmit2>&
+            Logger::Warn("GenerateLoaderType: DefinePinvokeMap failed");
+            return hr;
+        }
     }
 
-    // Create a string representing "OpenTelemetry.AutoInstrumentation.Loader.Loader"
-    LPCWSTR load_helper_str      = L"OpenTelemetry.AutoInstrumentation.Loader.Loader";
-    auto    load_helper_str_size = wcslen(load_helper_str);
+    // Add IL instructions into .cctor
+    {
+        // TypeRef/TypeDef for System.Byte
+        mdToken byte_type_token;
+        {
+            hr = resolver.GetTypeRefOrDefByName(corlib_ref, WStr("System.Byte"), &byte_type_token);
+            if (FAILED(hr))
+            {
+                Logger::Warn("GenerateLoaderType: GetTypeRefOrDefByName System.Byte failed");
+                return hr;
+            }
+        }
 
-    mdString load_helper_token;
-    hr = metadata_emit->DefineUserString(load_helper_str, (ULONG)load_helper_str_size, &load_helper_token);
-    if (FAILED(hr))
+        // MethodRef/MethodDef for
+        // void [mscorlib]System.Runtime.InteropServices.Marshal::Copy(native int, uint8[], int32, int32)
+        mdToken marshal_copy_member_ref;
+        {
+            // TypeRef/TypeDef for [mscorlib]System.Runtime.InteropServices.Marshal
+            mdTypeRef marshal_type_token;
+            hr = resolver.GetTypeRefOrDefByName(corlib_ref, WStr("System.Runtime.InteropServices.Marshal"),
+                                                &marshal_type_token);
+            if (FAILED(hr))
+            {
+                Logger::Warn("GenerateLoaderType: DefineTypeRefByName failed");
+                return hr;
+            }
+
+            COR_SIGNATURE marshal_copy_signature[] = {IMAGE_CEE_CS_CALLCONV_DEFAULT, // Calling convention
+                                                      4,                             // Number of parameters
+                                                      ELEMENT_TYPE_VOID,             // Return type
+                                                      ELEMENT_TYPE_I,                // List of parameter types
+                                                      ELEMENT_TYPE_SZARRAY,
+                                                      ELEMENT_TYPE_U1,
+                                                      ELEMENT_TYPE_I4,
+                                                      ELEMENT_TYPE_I4};
+            hr = resolver.GetMemberRefOrDef(marshal_type_token, WStr("Copy"), marshal_copy_signature,
+                                            sizeof(marshal_copy_signature), &marshal_copy_member_ref);
+            if (FAILED(hr))
+            {
+                Logger::Warn(
+                    "GenerateLoaderType: GetMemberRefOrDef System.Runtime.InteropServices.Marshal::Copy failed");
+                return hr;
+            }
+        }
+
+        // MethodRef/MethodDef for
+        // class [mscorlib]System.Reflection.Assembly [mscorlib]System.Reflection.Assembly::Load(uint8[], uint8[])
+        mdToken appdomain_load_member_ref;
```

Comment:
This could be a good time to make a small correction to the variable name, up to you

```suggestion
        mdToken assembly_load_member_ref;
```